### PR TITLE
dev/financial#74 Create ParticipantPayment record for every participant when registering multiple participants for event

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
We have `ParticipantPayment` and `MembershipPayment` records in CiviCRM which link a contribution to a participant / membership and show which contribution was used to "pay" for the participant/membership.  `ParticipantPayment` is not actually used very much, compared with `MembershipPayment` which is used for various UI forms when showing related memberships / contributions etc.
It was identified at the sprint in Barcelona that when registering multiple participants a `ParticipantPayment` record was only being created for the primary participant instead of for all participants.

Before
----------------------------------------
Only one participantPayment record created for multiple participant registration

After
----------------------------------------
One participantPayment record created for each participant in a multiple participant registration

Technical Details
----------------------------------------
The Order API creates one per participant. The MembershipPayment / ParticipantPayment records are not supposed to show who paid (that can be identified from the contribution) but is supposed to show that "this membership/participant" was paid for from this contribution.

Comments
----------------------------------------
@mecachisenros @eileenmcnaughton This is what we found at the sprint 
Thanks @sarvesh211999 for the unit test

### CODE REVIEW

**Comment by mattwire** [comment]
test this please

**Comment by seamuslee001** [comment]
@mattwire 

```
PHP Fatal error:  Cannot declare class CRM_Event_Form_Registration_ConfirmTest, because the name is already in use in /home/jenkins/bknix-dfl/build/core-15555-i9ur/web/sites/all/modules/civicrm/tests/phpunit/CRM/Event/Form/Registration/ConfirmMultipleParticipantTest.php on line 103
```

**Comment by eileenmcnaughton** [comment]
Looking at the code I find it more convincing that this was a bug since this appears to be fixing something that was broken rather than changing behaviour - it could even be a recent regression? We should double check when ``if (empty($ids['contribution'])) {``` stopped being true

**Comment by eileenmcnaughton** [comment]
Hmm - OK I looked & I think master is OK for this - I can't see evidence this was a recent regression - it probably regressed a while back but I now believe it IS a regression rather than a behaviour change

**Comment by seamuslee001** [comment]
@mattwire test failure looks related

File: tests/phpunit/CRM/Event/Form/Registration/ConfirmTest.php [hunk 1]
```diff
@@ -584,4 +584,89 @@ public function testNoteSubmission() {
...
+          'amount' => '200.00',
+          'tax_amount' => 20,
+        ],
+        [
+          'qfKey' => 'e6eb2903eae63d4c5c6cc70bfdda8741_2801',
+          'entryURL' => "http://dmaster.local/civicrm/event/register?reset=1&amp;id={$event['id']}",
+          'first_name' => 'Participant3',
+          'last_name' => 'LastName',
+          'email-Primary' => 'participant3@example.com',
+          'scriptFee' => '',
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Event/Form/Registration/ConfirmTest.php]: it's not a biggie but I've started taking out scriptFee & scriptArray from other tests as I've dug & concluded they should go entirely & are not used

Comment (mattwire) [reply tests/phpunit/CRM/Event/Form/Registration/ConfirmTest.php]: Removed - see https://lab.civicrm.org/dev/event/issues/19

**Comment by eileenmcnaughton** [comment]
@mattwire still related fails :-(

**Comment by eileenmcnaughton** [comment]
@mattwire still failing :-(

**Comment by mattwire** [comment]
Closing for now as there is no reasonable chance that I'm going to circle back to this any time soon. Its's being tracked via https://lab.civicrm.org/dev/financial/issues/74  @eileenmcnaughton if you get chance to take a look at any of this, it already has a test here :-) The issue I hit is a rabbit hole of converting follow-on code (ie. receipting code etc) to read from an array instead of an integer variable. I know you've done some cleanup so it might be easier now.

**Pull request closed without merging**